homeassistant/components/heos/config_flow.py

```diff
@@ -74,3 +81,24 @@ async def async_step_user(
             data_schema=vol.Schema({vol.Required(CONF_HOST, default=host): host_type}),
             errors=errors,
         )
+
+    async def async_step_reconfigure(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Allow reconfiguration of entry."""
+        await self.async_set_unique_id(DOMAIN)
+
+        entry = self._get_reconfigure_entry()
+        host = entry.data[CONF_HOST]  # Get current host value
+        errors: dict[str, str] = {}
+        if user_input is not None:
+            host = user_input[CONF_HOST]
+            if await _validate_host(host, errors):
+                return self.async_update_reload_and_abort(
+                    entry, data_updates={CONF_HOST: host}
+                )
+        return self.async_show_form(
+            step_id="reconfigure",
+            data_schema=vol.Schema({vol.Required(CONF_HOST, default=host): str}),
```

Comment:
```suggestion
            data_schema=vol.Schema({vol.Required(CONF_HOST, default=entry.data[CONF_HOST]): str}),
```

Reply:
ah right, that's what you were using this for 👍

Reply:
I'll add a line to the tests to validate the behavior. :)